packages/sdk/src/swap/services/ApiService.ts

```diff
@@ -102,26 +103,29 @@ const getQuote: BackendQuery<QuoteRequest, QuoteResponse> = async (
 
   const { data } = await axios.get<QuoteQueryResponse>(url, { signal });
 
-  return { quote: data, ...quoteRequest };
+  return { ...quoteRequest, quote: data };
 };
 
-const requestDepositAddress: BackendQuery<SwapRequest, SwapResponse> = async (
-  baseUrl,
-  route,
-  { signal },
-) => {
+const requestDepositAddress: BackendQuery<
+  DepositAddressRequest,
+  DepositAddressResponse
+> = async (baseUrl, swapRequest, { signal }) => {
   const body: SwapRequestBody = {
-    destAddress: route.destAddress,
-    srcAsset: route.srcAsset,
-    destAsset: route.destAsset,
-    amount: route.amount,
+    destAddress: swapRequest.destAddress,
+    srcAsset: swapRequest.srcAsset,
+    destAsset: swapRequest.destAsset,
+    amount: swapRequest.amount,
   };
 
   const url = new URL('/swaps', baseUrl).toString();
 
-  const { data } = await axios.post<SwapResponse>(url, body, { signal });
+  const { data } = await axios.post<PostSwapResponse>(url, body, { signal });
 
-  return data;
+  return {
+    ...swapRequest,
+    depositChannelId: data.id,
```

Comment:
this is a breaking change. but i think its important to make it explicit what kind of id this is